### ISSUE TITLE
Rename "Tones" to "Levels"?

### ISSUE THREAD

**Issue by unfa** [RodZill4/material-maker#194]
The "Tones" nodes are doing what most graphics software would call "levels" instead.

![image](https://user-images.githubusercontent.com/491704/96034430-d39f9680-0e61-11eb-990e-638d8be6d276.png)

Here's an example from GIMP:

![image](https://user-images.githubusercontent.com/491704/96034558-fe89ea80-0e61-11eb-9be6-cc833b6a341f.png)

Also - a similar node in Substance Designer:
https://docs.substance3d.com/sddoc/levels-172825279.html

What's funny - the tooltip on the Tones node mentiones levels:
![image](https://user-images.githubusercontent.com/491704/96034684-31cc7980-0e62-11eb-9e22-0a0c70be2d14.png)

I'd recommend renaming these nodes as follows:

Tones → Levels
Tones map → Levels Mapping
Tones step → Levels Sweep

What do you think?

**Comment by RodZill4** [comment]
"The Level node allows you to remap the **tones** of an input..." (from the doc of the "Levels" node in SD's doc).

"Level" is a very generic word ("a position on a scale of amount, quantity, extent, or quality" in dictionary). The Gimp name is "color levels" which is accurate. As a consequence, "level" is a good name for the **parameters** of that node (hence the tooltip).

I don't think copying names from SD is a good idea (especially for "Alveolus" :D). I'd rather not do it to be honest, I don't want MM to be perceived as a "free clone of SD".

But it would probably be a good idea to have a mechanism that pops the "good" node when you search (in library or node popup) the name of the node in SD.

**Comment by Jummit** [comment]
I have to admit I always search for levels, and then have to remember what it is called in MaterialMaker. I'd go with the industry standard name on this one.

**Comment by RodZill4** [comment]
You can still create a copy of those nodes into the user library (library management will be improved in next release, but it's quite easy in 0.93).

**Comment by Calinou** [comment]
> But it would probably be a good idea to have a mechanism that pops the "good" node when you search (in library or node popup) the name of the node in SD.

This can be done by adding an "aliases" field (which can be a comma-separated string or an array) to each node. This field wouldn't be displayed on the user interface, but would be used in addition to the description for the purposes of searching.

**Comment by unfa** [comment]
Ah, the Tones Step is an equivalent of SD's Histogram Scan. I misremembered "Sweep".

I'm generally don't want to be pushing towards making MM a clone of SD, thought "Levels" is pretty standard in any software, so I think it's a good convention to follow. We're not gonna remap Ctrl+S to something else just because SD uses this shortcut as well right?

I mean - being different just for the sake of being different is generally only achieving one thing - making it harder for new users to learn the software. Blender has finally learned it's lesson there and adopted  bunch of industry standards in 2.80, and I think that's great.

Also - honestly "Tones" in my mind is related to "Tonemapping" - and I think about Filmic when I hear that.
Levels is what I'm searching for to do this kind of stuff.

Also - I think making aliases that will give users a correct node when they type in a name from SD or Blender or Nuke/Natron could be a very good idea. I'm all for it, as it'd flatten the learning curve without giving up MM's unique take.

Though I'd really prefer Tones to be renamed to Levels :D
It feels to me as if "Blur" was called "Spatial Smooth" instead.

**Comment by RodZill4** [comment]
Hmm to be clear, here's how I proceed when naming nodes:
- if the name is "good" or if I have no better name, I keep it. For example "Blur" in SD has a good name, because that's how the operation is named in a dictionary. "Slope Blur" is not a good name but it's difficult to find an accurate name that is short, so I kept it.
- if the name is "bad" (not accurate enough) and I believe have a "better" name I change it. For example "Flood fill" became "Fill" because "Fill" is the name of the operation the node performs, and "Flood fill" is the name of an algorithm. And (1) the user does not care about the algorithm, and (2) I did not use that algorithm.

"Tones" related to "Tonemapping"? Yes, that's exactly what this node does (but just not in the "Filmic" sense).

Anyway, if enough users ask for that change, it will definitely happen (I'd say I don't care, I always use Colorize instead, much more flexible :p).
